waspc/src/Wasp/Node/Version.hs

```diff
@@ -39,79 +40,101 @@ data VersionCheckResult
-      "Wasp requires Node version " <> show oldestWaspSupportedNodeVersion <> " or higher."
-    ]
-
--- | Gets the installed node version, if any is installed, and returns it.
---
--- Returns a string representing the error condition if node's version could
--- not be found.
-getUserNodeVersion :: IO (Either ErrorMessage SV.Version)
-getUserNodeVersion = do
-  -- Node result is one of:
-  -- 1. @Left processError@, when an error occurs trying to run the process
-  -- 2. @Right (ExitCode, stdout, stderr)@, when the node process runs and terminates
-  nodeResult <-
-    (Right <$> P.readProcessWithExitCode "node" ["--version"] "")
-      `catchIOError` ( \e ->
-                         if isDoesNotExistError e
-                           then return $ Left nodeNotFoundMessage
-                           else return $ Left $ makeNodeUnknownErrorMessage e
-                     )
-  return $ case nodeResult of
-    Left procErr ->
-      Left
-        ( unlines
-            [ "Running `node --version` failed.",
-              indent 2 procErr,
-              "Make sure you have `node` installed and in your PATH."
-            ]
-        )
-    Right (ExitFailure code, _, stderr) ->
-      Left
-        ( unlines
-            [ "Running `node --version` failed (exit code " ++ show code ++ "):",
-              indent 2 stderr
-            ]
-        )
+getAndCheckUserNodeVersion = checkNodeVersion getNodeVersion
+  where
+    checkNodeVersion = checkInstalledVersionNewerThanOldestSupported "node" oldestWaspSupportedNodeVersion
+    getNodeVersion = parseVersionFromCommandOutput "node" ["--version"]
+
+getAndCheckUserNPMVersion :: IO VersionCheckResult
+getAndCheckUserNPMVersion = checkNPMVersion getNPMVersion
+  where
+    checkNPMVersion = checkInstalledVersionNewerThanOldestSupported "npm" oldestWaspSupportedNPMVersion
+    getNPMVersion = parseVersionFromCommandOutput "npm" ["--version"]
+
+checkInstalledVersionNewerThanOldestSupported :: String -> SV.Version -> IO (Either ErrorMessage SV.Version) -> IO VersionCheckResult
+checkInstalledVersionNewerThanOldestSupported commandName oldestSupportedVersion getInstalledVersion = do
+  result <- getInstalledVersion
+  return $ case result of
+    Left errorMsg -> VersionCheckFail errorMsg
+    Right userVersion ->
+      if SV.isVersionInRange userVersion $ SV.Range [SV.gte oldestSupportedVersion]
+        then VersionCheckSuccess
+        else VersionCheckFail $ versionMismatchErrorMessage userVersion
+  where
+    versionMismatchErrorMessage :: SV.Version -> ErrorMessage
+    versionMismatchErrorMessage userVersion =
+      unlines
+        [ "Your " ++ commandName ++ " version does not meet Wasp's requirements! You are running " ++ commandName ++ " " <> show userVersion <> ".",
+          "Wasp requires " ++ commandName ++ " version " <> show oldestSupportedVersion <> " or higher."
+        ]
+
+parseVersionFromCommandOutput :: String -> [String] -> IO (Either ErrorMessage SV.Version)
```

Comment:
A bit wrong name, it doesn't sound like you are also calling the command, while you are.

Reply:
Actually might be fine, it is not so easy to come with a better name, and it is relatively clear what is happening if you look at the arguments. So I am good with the current name if you don't have something clearly better.